### ISSUE TITLE
Can't use emit matcher when deploying contracts

### ISSUE THREAD

**Issue by alcuadrado** [EthWorks/Waffle#196]
Hi,

I can't find a way to use `await expect(...).to.emit(...)` when deploying a contract, as I have to pass it the `Contract` instance.

Am I missing something? Is this a known limitation?

I'm deploying with `ContractFactory.deploy(...)`, but I don't think that's the problem.

**Comment by sz-piotr** [comment]
Yeah, the current `.to.emit` is very inflexible. The problem is that we want to combine a nice api with filtering logs which results in things like this.

How would you like `.to.emit` to work in this case?
